### ISSUE TITLE
Unable to replicate PCC drop in dpr-reader-single-nq-base linear layer using ttnn.matmul and ttnn.add

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-forge-fe#2604]
## Description

- The `dpr-reader-single-nq-base` model is encountering the following tensor mismatch at runtime:

```
Tensor mismatch. Required rtol=1e-05, atol=1e-08
Observed maximum relative diff: 0.004826769698411226, maximum absolute diff: 0.0004223063588142395
Golden: (shape = torch.Size([])
tensor(0.0871, grad_fn=<SelectBackward0>)
Calculated: (shape = torch.Size([])
tensor(0.0875)
```
- The drop originates from the this [linear layer](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/models/dpr/modeling_dpr.py#L246C28-L246C46) 
- In [nn.linear]( https://docs.pytorch.org/docs/stable/generated/torch.nn.Linear.html), output is computed as: `output=X@(transpose of W) +b` Where:
  - X is the input tensor of shape (batch_size,in_features)(batch_size,in_features)
  - W is the weight matrix of shape (out_features,in_features)(out_features,in_features)
  - b is the bias of shape (out_features)(out_features)

- **Unable to narrow down the drop to specific op in nn.linear as below matmul & add cases are passed in tt-metal**

## cases

```
def test_add_my_case(device):
    torch_input_tensor_a = torch.randn(1,1)
    torch_input_tensor_b = torch.randn(1,1)
    torch_output_tensor = torch.add(torch_input_tensor_a, torch_input_tensor_b)

    input_tensor_a = ttnn.from_torch(torch_input_tensor_a, layout=ttnn.TILE_LAYOUT, device=device)
    input_tensor_b = ttnn.from_torch(torch_input_tensor_b, layout=ttnn.TILE_LAYOUT, device=device)
    output = ttnn.add(input_tensor_a, input_tensor_b)
    output = ttnn.to_torch(output)

    assert_with_pcc(torch_output_tensor, output)


def test_matmul_my_case(device):
    x = torch.randn(1, 768)
    y = torch.randn(768, 1)
    x_tt = ttnn.from_torch(x, layout=ttnn.TILE_LAYOUT, device=device)
    y_tt = ttnn.from_torch(y, layout=ttnn.TILE_LAYOUT, device=device)
    z_tt = ttnn.matmul(x_tt, y_tt)
    z = ttnn.to_torch(z_tt)
    z_t = torch.matmul(x, y)
    assert_with_pcc(z_t, z)
```

## Reproduce

```
git checkout kkannan/jul18_dpr_pcc_drop
git submodule update --init --recursive

# Model run 
pytest forge/test/models/pytorch/text/dpr/test_dpr.py::test_dpr_reader_pytorch[facebook/dpr-reader-single-nq-base] -svv 

# Model specific sanity
pytest forge/test/models/pytorch/text/dpr/test_s1.py

# op specific sanity
pytest forge/test/models/pytorch/text/dpr/test_s2.py

In tt-metal,

git checkout kkannan/jul18_dpr_matmul_add_metal
git submodule update --init --recursive

# matmul
pytest tests/ttnn/unit_tests/operations/matmul/test_matmul.py -svv -k "my"

# add 
pytest tests/ttnn/unit_tests/operations/eltwise/test_add.py -svv -k "my" 

```

## Observed behaviour 

Model run 
```
2025-07-18 09:14:23.897 | ERROR    | forge.verify.compare:compare_with_golden:43 - Tensor mismatch. Required rtol=1e-05, atol=1e-08
2025-07-18 09:14:23.897 | ERROR    | forge.verify.compare:compare_with_golden:44 - Observed maximum relative diff: 0.2652907967567444, maximum absolute diff: 0.26116979122161865
2025-07-18 09:14:23.897 | ERROR    | forge.verify.compare:compare_with_golden:49 - Golden: (shape = torch.Size([])
2025-07-18 09:14:23.898 | ERROR    | forge.verify.compare:compare_with_golden:50 - tensor(-1.2456, grad_fn=<SelectBackward0>)
2025-07-18 09:14:23.899 | ERROR    | forge.verify.compare:compare_with_golden:51 - Calculated: (shape = torch.Size([])
2025-07-18 09:14:23.901 | ERROR    | forge.verify.compare:compare_with_golden:52 - tensor(-0.9845)
```

Model specific sanity
```
2025-07-18 13:42:48.537 | ERROR    | forge.verify.compare:compare_with_golden:44 - Observed maximum relative diff: 0.002990416483953595, maximum absolute diff: 0.0007980167865753174
2025-07-18 13:42:48.537 | ERROR    | forge.verify.compare:compare_with_golden:49 - Golden: (shape = torch.Size([])
2025-07-18 13:42:48.538 | ERROR    | forge.verify.compare:compare_with_golden:50 - tensor(-0.2661, grad_fn=<SelectBackward0>)
2025-07-18 13:42:48.539 | ERROR    | forge.verify.compare:compare_with_golden:51 - Calculated: (shape = torch.Size([])
2025-07-18 13:42:48.539 | ERROR    | forge.verify.compare:compare_with_golden:52 - tensor(-0.2669)
```

op specific sanity
```
2025-07-18 13:44:04.561 | ERROR    | forge.verify.compare:compare_with_golden:43 - Tensor mismatch. Required rtol=1e-05, atol=1e-08
2025-07-18 13:44:04.561 | ERROR    | forge.verify.compare:compare_with_golden:44 - Observed maximum relative diff: 0.004826769698411226, maximum absolute diff: 0.0004223063588142395
2025-07-18 13:44:04.561 | ERROR    | forge.verify.compare:compare_with_golden:49 - Golden: (shape = torch.Size([])
2025-07-18 13:44:04.561 | ERROR    | forge.verify.compare:compare_with_golden:50 - tensor(0.0871, grad_fn=<SelectBackward0>)
2025-07-18 13:44:04.562 | ERROR    | forge.verify.compare:compare_with_golden:51 - Calculated: (shape = torch.Size([])
2025-07-18 13:44:04.562 | ERROR    | forge.verify.compare:compare_with_golden:52 - tensor(0.0875)

# matmul 

PASSED

# add

PASSED

```

## Logs

model run - [jul18_dpr.log](https://github.com/user-attachments/files/21317842/jul18_dpr.log)
Model specific sanity - [jul18_z_s1.log](https://github.com/user-attachments/files/21317845/jul18_z_s1.log)
op specific sanity - [jul18_z_s2.log](https://github.com/user-attachments/files/21317846/jul18_z_s2.log)
matmul - [jul18_matmul.log](https://github.com/user-attachments/files/21318928/jul18_matmul.log)
add -  [jul18_add.log](https://github.com/user-attachments/files/21318930/jul18_add.log)

**Comment by kamalrajkannan78** [comment]
While running the unique op tests generated by `scripts/model_analysis/generate_models_ops_test.py` on model a tensor mismatch was observed in matmul and index ops. [aug1_dpr_op_run.log](https://github.com/user-attachments/files/21547146/aug1_dpr_op_run.log)

```
FAILED forge/test/models/pytorch/text/dpr_debug/dpr_ops/test_index.py::test_module[Index3-[((2,), torch.float32)]]
FAILED forge/test/models/pytorch/text/dpr_debug/dpr_ops/test_matmul.py::test_module[Matmul0-[((1, 768), torch.float32), ((768, 1), torch.float32)]]
```

While running the unique op tests generated by `scripts/model_analysis/generate_models_ops_test.py` on op specific sanity a tensor mismatch was observed in matmul. [aug1_dpr_s_op_run.log](https://github.com/user-attachments/files/21547387/aug1_dpr_s_op_run.log)

```
FAILED forge/test/models/pytorch/text/dpr_sanity/dpr_s_ops/test_matmul.py::test_module[Matmul0-[((1, 768), torch.float32), ((768, 1), torch.float32)]]
```

This indicates that the accuracy drop originates from the ttnn.matmul & ttnn.slice (index operation is mapped to ttnn.slice) op. The issue was further cross-verified by running the problematic case on Metal, where the drop was also observed


### Repro

```
git checkout kkannan/aug1_dpr_debug 
git submodule update --recursive
git lfs pull --include 'index_input_0.pt matmul_input_0.pt matmul_input_1.pt' 
```

To generate unique op tests for model & sanity. [aug1_dpr_op_gen.log](https://github.com/user-attachments/files/21547346/aug1_dpr_op_gen.log) , [aug1_dpr_s_op_gen.log](https://github.com/user-attachments/files/21547345/aug1_dpr_s_op_gen.log)

```
python scripts/model_analysis/generate_models_ops_test.py --test_directory_or_file_path forge/test/models/pytorch/text/dpr_debug/ --models_ops_test_output_directory_path forge/test/models/pytorch/text/dpr_debug/ --models_ops_test_package_name dpr_ops -m "not skip_model_analysis" 
```

Delete models_unique_ops_output/ then

```
python scripts/model_analysis/generate_models_ops_test.py --test_directory_or_file_path forge/test/models/pytorch/text/dpr_sanity/ --models_ops_test_output_directory_path forge/test/models/pytorch/text/dpr_sanity/ --models_ops_test_package_name dpr_s_ops -m "not skip_model_analysis" 
```

To run unique op tests generated for model and sanity

```
pytest forge/test/models/pytorch/text/dpr_debug/dpr_ops/ -svv 
pytest forge/test/models/pytorch/text/dpr_sanity/dpr_s_ops/ -svv

```

### Additional logs 

[aug1_dpr_org.log](https://github.com/user-attachments/files/21547393/aug1_dpr_org.log) ( whole model run mismatch)
